draft-ietf-anima-brski-prm.md

```diff
@@ -1238,406 +1298,349 @@ In {{!RFC8995}}, the pledge generates PVR as CMS-signed JSON and PER as PKCS#10
+Note that BRSKI-PRM allows for alternative formats such as CMS-signed JSON as used in BRSKI {{!RFC8995}} or COSE-signed CBOR for constrained environments, when defined by other specifications.
+Overall, a MASA responsible for BRSKI-PRM capabable pledges MUST support the same formats as supported by those pledges.
 
-* 406 Not Acceptable: if the Content-Type indicated by the Accept header is unknown or unsupported.
+Once the MASA receives the RVR artifact, it MUST perform the verification as described in {{Section 5.5 of !RFC8995}}.
+Depending on policy, the MASA MAY chose the kind of assertion to perform.
+For the Agent-Proximity Assertion of BRSKI-PRM (see {{agt_prx}}), the MASA MUST skip the verification described in {{Section 5.5.5 of !RFC8995}} and instead MUST verify for the PVR contained in the `prior-signed-voucher-request` field of the RVR that
 
-If the validation succeeds, the registrar performs pledge authorization according to {{Section 5.3 of !RFC8995}} followed by obtaining a voucher from the pledge's MASA according to {{Section 5.4 of !RFC8995}} with the modifications described below in {{rvr-proc}}.
+* the `agent-provided-proximity-registrar-cert` field contains the registrar LDevID certificate and corresponds to the LDevID credentials used to sign the RVR;
+  TODO(unclear): Note: Correspond here relates to the case that a single registrar LDevID certificate is used or that different registrar LDevID certificates are used, which are issued by the same CA.
+* the `agent-signed-data` field is signed with the credentials corresponding to the Registrar-Agent LDevID certificate provided in the `agent-sign-cert` field of the RVR;
+  this is done via the SubjectKeyIdentifier of the certificate in the `kid` Header Parameter of the JWS Protected Header in the `agent-signed-data` field;
+* the registrar LDevID certificate and the Registrar-Agent LDevID certificate belong to the same domain (i.e., have a common domain CA certificate in their chains).
+* the product-serial-number inside the `agent-signed-data` is equal to the `serial-number` field of PVR and the `serial-number` field of the RVR as well as the X520SerialNumber field of the pledge IDevID certificate, which is contained in the JWS Protected Header of the PVR.
 
-### Supply RVR to MASA (backend interaction) {#rvr-proc}
+If the `agent-sign-cert` field in the RVR is not set, the MASA MAY state a lower level assertion value instead of failing the verification, e.g., "logged" or "verified".
 
-The registrar needs to convert the PVR to an RVR and supply it to the MASA.
+If the verification fails, the MASA SHOULD respond with an HTTP client error status code to the registrar.
+The client error status codes are kept the same as defined in {{Section 5.6 of !RFC8995}}:
 
-If the MASA address/URI is learned from the IDevID MASA URI extension ({{Section 2.3 of !RFC8995}}), then the MASA on that URI MUST support the procedures defined in this document if the PVR used JSON-JWS encoding.
```

Comment:
Simplified by MASA expectation to support BRSKI-PRM when implemented by pledges. Statement in arch overview and where it happens in the flow.

Reply:
okay